### PR TITLE
WIP Manual 2.1.x updates

### PR DESCRIPTION
**Pull request by esbrandt**
This PR aims to document the following user visible changes, merged to https://github.com/mixxxdj/mixxx/master since the latest stable release.

This is my filter on https://github.com/mixxxdj/mixxx/pulls
 
  ``is:pr is:closed merged:>=2015-12-29 sort:updated-asc``

TODO
=====
- [x] Highlight Crates a Track is in
  https://github.com/mixxxdj/mixxx/pull/617
- [x] Write audio tags back into files
  https://github.com/mixxxdj/mixxx/pull/728
- [x] Print duration of selected tracks in AutoDJ
  https://github.com/mixxxdj/mixxx/pull/749
- [x] Added multi import for crates and playlists 
  https://github.com/mixxxdj/mixxx/pull/945
- [ ] add option to generate waveforms on analysis
  https://github.com/mixxxdj/mixxx/pull/998
- [x] sort keys by circle of fifths
  https://github.com/mixxxdj/mixxx/pull/1000
- [ ] Add support for vertical waveforms and overviews
  https://github.com/mixxxdj/mixxx/pull/983
- [x] Add Cue labels
  https://github.com/mixxxdj/mixxx/pull/992
- [x] Add BPM conversions 4/3 and 3/2  
  https://github.com/mixxxdj/mixxx/pull/1106
- [ ] Broadcast reconnect
  https://github.com/mixxxdj/mixxx/pull/1031
- [x] Outgoing MIDI debugging messages 
  https://github.com/mixxxdj/mixxx/pull/1004
- [x] Add an option to replace the autoDJ queue with the selected tracks 
  https://github.com/mixxxdj/mixxx/pull/1163
- [ ] ~~new Deere effects units with meta knobs~~
  https://github.com/mixxxdj/mixxx/pull/1063
- [ ] ~~Loudness Contour effect~~ 
  https://github.com/mixxxdj/mixxx/pull/1141
- [x] Command line option --debugAssertBreak
  https://github.com/mixxxdj/mixxx/pull/1189
- [ ] ~~Biquad EQ with full kill feature~~ 
  https://github.com/mixxxdj/mixxx/pull/1164
- [ ] ~~Effect selector widget~~
  https://github.com/mixxxdj/mixxx/pull/1135 
- [ ] ~~add per-effect meta knobs~~
  https://github.com/mixxxdj/mixxx/pull/1062
- [x] Make the overview-waveform darker for the played portion of the track 
  https://github.com/mixxxdj/mixxx/pull/1203
- [x] Recording duration widget
  https://github.com/mixxxdj/mixxx/pull/1221
- [ ] Keylock: Option how key is unlocked/reset
  https://github.com/mixxxdj/mixxx/pull/1222
- [ ] Auto dj crates improvements
  https://github.com/mixxxdj/mixxx/pull/1214
- [ ] Various fixes and additions related to recording the mix to a file
  https://github.com/mixxxdj/mixxx/pull/1035
- [ ] HiDPI Skin Scaling
  https://github.com/mixxxdj/mixxx/pull/1204
- [x] add option to reset deck gain on track load
  https://github.com/mixxxdj/mixxx/pull/1250
- [x] option to show both elapsed and remaining time simultaneously
  https://github.com/mixxxdj/mixxx/pull/1216
- [ ] Tango skin
  https://github.com/mixxxdj/mixxx/pull/1151
- [ ] new loop and beatjump controls 
  https://github.com/mixxxdj/mixxx/pull/1187
- [ ] Library navigation controls 
  https://github.com/mixxxdj/mixxx/pull/953
- [ ] Deere 2.1
  https://github.com/mixxxdj/mixxx/pull/940
- [x] Add command for clearing waveform to track's context menu
  https://github.com/mixxxdj/mixxx/pull/1197
- [ ] Screensaver inhibitor
  https://github.com/mixxxdj/mixxx/pull/1229
- [x] Crate filter
  https://github.com/mixxxdj/mixxx/pull/1263
- [x] Fix and improve Logging & Debugging
  https://github.com/mixxxdj/mixxx/pull/1265
- [x] Use checkbox menu in crate selection
  https://github.com/mixxxdj/mixxx/pull/1341
- [x] Clean up track right click context menu
  https://github.com/mixxxdj/mixxx/pull/1427
- [x] --logFlushLevel Define which messages are written to mixxx.log
   https://github.com/mixxxdj/mixxx/pull/1483

Note:
I will not cover effects in this PR (~~crossed out bullet points~~), since #64 is already in progress

### CODE REVIEW

**Comment by Be-ing** [comment]
Thanks for getting started on this!

**Comment by Be-ing** [comment]
Because of the very long time between branching 1.12 and releasing it, there were a lot of PRs merged to master before the 2.0 release that were not in the 2.0 release. So we actually need to look back further.

File: source/chapters/configuration.rst [hunk 1]
```diff
@@ -267,32 +267,48 @@ Equalizer Preferences
...
 
 * **High/Low Shelf EQ**: This slider sets the crossover frequencies of the mixing
   EQ. It controls which frequency range is affected by the low, mid, and high
   channel EQ knobs. By default the low knob controls the bass and sub bass
   range up to 246 Hz. The mid knob controls the mid range up to 2.5 kHz.
   The remaining treble range is controlled by the high knob.
 
-* **Master EQ**: This section allows you to setup an EQ that affects the master
-  output.
+* **Master EQ**: This section allows you to setup an graphic equalizer that affects
```
Comment (Be-ing) [inline source/chapters/configuration.rst]: typo: "a graphic equalizer"

**Comment by Be-ing** [comment]
mixxxdj/mixxx#983 did not make any user facing changes because no one has published a skin using vertical waveforms (yet).

**Comment by Be-ing** [comment]
This has some merge conflicts since #60 was merged. @esbrandt could you resolve those?

**Comment by Be-ing** [comment]
In general, I do not think we should document *everything* in the manual. IMO the manual should only be used for explaining what would take too many words to fit into the user interface. Everything else should be self-explanatory from the design of the UI or explained in tooltips. If we clutter the manual with information that is obvious just from using Mixxx, it makes it harder to find information that users do need from the manual.

File: source/_templates/layout.html [hunk 2]
```diff
@@ -5,7 +5,7 @@
     Master layout for Mixxx manual. Adapted from Basic theme.
 
     :copyright: Copyright 2007-2011 by the Sphinx team, see AUTHORS.
-    :copyright: Copyright 2011 by the Mixxx Team
+    :copyright: Copyright 2011-2017 by the Mixxx Development Team
```
Comment (daschuer) [inline source/_templates/layout.html]: 2018?

File: source/chapters/appendix.rst [hunk 3]
```diff
@@ -263,22 +248,22 @@ Mixxx is designed to be as user-friendly as possible. As such, its command line
 options are only useful for development or debugging, as they make these tasks
 easier. Command line options are case-sensitive.
 
-.. versionadded:: 2.0
+.. versionadded:: 2.1.0
+
+   * ``--debugAssertBreak`` option
+   * ``--logLevel`` option
```
Comment (daschuer) [inline source/chapters/appendix.rst]: Please Add: 

``` ``--logFlushLevel`` option ```

File: source/chapters/appendix.rst [hunk 4]
```diff
@@ -311,10 +297,26 @@ Option                  Description
...
 ----------------------  -------------------------------------------------
 -f, --fullScreen        Starts Mixxx in full-screen mode
 ----------------------  -------------------------------------------------
+--logLevel LEVEL        Sets the verbosity of command line logging
+
+                        * ``critical`` - Critical/Fatal only
+                        * ``warning``  - Above + Warnings
+                        * ``info``     - Above + Informational messages
+                        * ``debug``    - Above + Debug/Developer messages
+                        * ``trace``    - Above + Profiling messages
```
Comment (daschuer) [inline source/chapters/appendix.rst]: --logFlushLevel LEVEL   

Sets the the logging level at which the log buffer is flushed to mixxx.log. LEVEL is one of the values defined at --logLevel above.

File: source/conf.py [hunk 5]
```diff
@@ -49,7 +49,7 @@
 
 # General information about the project.
 project = u'Mixxx'
-copyright = u'2011-2015, The Mixxx Development Team'
+copyright = u'2011-2017, The Mixxx Development Team'
```
Comment (daschuer) [inline source/conf.py]: 2018?

File: source/conf.py [hunk 6]
```diff
@@ -286,7 +286,7 @@
 epub_title = u'Mixxx User Manual'
 epub_author = u'The Mixxx Development Team'
 epub_publisher = u'The Mixxx Development Team'
-epub_copyright = u'2011-2015, The Mixxx Development Team'
+epub_copyright = u'2011-2017, The Mixxx Development Team'
```
Comment (daschuer) [inline source/conf.py]: 2018

File: source/locale/pot/manual_guidelines.pot [hunk 7]
```diff
@@ -1,286 +1,286 @@
 # SOME DESCRIPTIVE TITLE.
-# Copyright (C) 2011-2015, The Mixxx Development Team
+# Copyright (C) 2011-2017, The Mixxx Development Team
```
Comment (daschuer) [inline source/locale/pot/manual_guidelines.pot]: 2018

**Comment by esbrandt** [comment]
skip ci

**Comment by esbrandt** [comment]
> skip ci

Manually solving merge conflicts. This takes some time (hopefully tonight), since the structure has changed quite a bit since this was started.

File: source/chapters/djing_with_mixxx.rst [hunk 8]
```diff
@@ -369,8 +366,17 @@ choose to create them in the preferences.
 Record your mix to disk
 -----------------------
 
-.. versionadded:: 2.0
-   Displays the duration of the recording.
+.. versionadded:: 2.10
```
Comment (rryan) [inline source/chapters/djing_with_mixxx.rst]: 2.1.0 ?  We aren't to 2.10 ... yet! :)

File: source/chapters/getting_involved.rst [hunk 9]
```diff
@@ -39,13 +41,13 @@ As a software developer
 * You don't have to know C++. There are developers who got into Mixxx
   development while learning C++ along the way.
 * Join our mailing list: `mixxx-devel
-  <https://lists.sourceforge.net/lists/listinfo/mixxx-devel>`_
-* Join our :term:`IRC` channel, #mixxx on `Freenode <http://www.freenode.net/>`_
+  <https://sourceforge.net/projects/mixxx/lists/mixxx-devel>`_
+* Join our :term:`IRC` channel, #mixxx on `Freenode <https://freenode.net/>`_
```
Comment (rryan) [inline source/chapters/getting_involved.rst]: Replace IRC with Zulip?

File: source/chapters/library.rst [hunk 10]
```diff
@@ -409,6 +420,10 @@ Edit metadata of audio files
 .. sectionauthor::
    S.Brandt <s.brandt@mixxx.org>
 
+.. versionadded:: 2.10
```
Comment (rryan) [inline source/chapters/library.rst]: 2.1.0

File: source/chapters/library.rst [hunk 11]
```diff
@@ -531,12 +550,19 @@ mixing or for using Mixxx as media player.
 
 The Auto DJ features in detail:
 
+.. versionadded:: 2.10
```
Comment (rryan) [inline source/chapters/library.rst]: 2.1.0

File: source/chapters/library.rst [hunk 12]
```diff
@@ -691,25 +719,22 @@ If however you want to plan out a specific set and practice the transitions,
 you will want an ordered list of tracks. Playlists serve this purpose.
 
 
-.. _library-browse:
+.. _library-computer:
 
-Browse - Loading remote tracks
-==============================
+Computer - Loading remote tracks
```
Comment (rryan) [inline source/chapters/library.rst]: "remote" is an odd word here. What about "Loading tracks from anywhere on your computer" ?

File: source/chapters/user_interface.rst [hunk 13]
```diff
@@ -56,15 +56,12 @@ Track Information Display
...
-.. versionadded:: 2.0
-   Parse the title and the artist from the file name when the corresponding
-   :term:`metadata` is missing.
-
 **Track Title**
   The title of the track that was loaded into a deck is displayed on top. This
   is the same as the title listed under the :guilabel:`Title` column heading in
   the Mixxx library. This information is initially loaded from the track's
-  :term:`metadata`.
+  :term:`metadata`. Is the metadata missing, parse the title and the artist
```
Comment (rryan) [inline source/chapters/user_interface.rst]: "If the metadata is missing, the title and artist are guessed from the filename."

File: source/chapters/user_interface.rst [hunk 14]
```diff
@@ -135,8 +138,9 @@ separate waveforms. Select your preferred appearance in
    Mixxx alternative skin (Deere) - Separate waveforms
 
 .. warning :: If you have a slower computer and notice performance issues with
-              Mixxx, try to lower the frame rate or change the waveform type in
-              :menuselection:`Preferences --> Waveforms`.
+              Mixxx, try to lower the frame rate or change the type for waveform
```
Comment (rryan) [inline source/chapters/user_interface.rst]: Flows a little better?

"try lowering the frame rate or changing the waveform type in :menuselection:`Preferences --> Interface --> Waveforms"

File: source/chapters/djing_with_mixxx.rst (hunk 8 above)
Comment (Be-ing) [reply source/chapters/djing_with_mixxx.rst]: IMO these "version added" notices clutter the manual. They're good for release announcements, but I don't know why someone reading the manual would care when a feature was added.

**Comment by esbrandt** [comment]
> It looks like there are still some merge conflicts.

New conflicts after yesterdays #64 merge, will have a look.

File: Makefile [hunk 15]
```diff
@@ -3,7 +3,7 @@
 
 # You can set these variables from the command line.
 SPHINXOPTS    =
-SPHINXBUILD   = sphinx-build-3
+SPHINXBUILD   = sphinx-build
```
Comment (Be-ing) [inline Makefile]: Whoops, sorry that slipped into a commit.

**Comment by Be-ing** [comment]
Thanks! I think we're good to go for the 2.1 release. Now that we have continuous deployment for the manual, we can continue to work on it after the release. There is always more work to do, but this is a good state for packaging the PDF for the release.

**Pull request merged**